src/NexusMods.Collections/InstallCollectionDownloadJob.cs

```diff
@@ -0,0 +1,321 @@
+using System.Collections.Concurrent;
+using System.IO.Hashing;
+using System.Security.Cryptography;
+using JetBrains.Annotations;
+using NexusMods.Abstractions.Collections.Types;
+using NexusMods.Abstractions.Collections.Json;
+using NexusMods.Abstractions.GameLocators;
+using NexusMods.Abstractions.IO;
+using NexusMods.Abstractions.IO.StreamFactories;
+using NexusMods.Abstractions.Jobs;
+using NexusMods.Abstractions.Library;
+using NexusMods.Abstractions.Library.Models;
+using NexusMods.Abstractions.Loadouts;
+using NexusMods.Abstractions.MnemonicDB.Attributes;
+using NexusMods.Abstractions.NexusModsLibrary;
+using NexusMods.Abstractions.NexusModsLibrary.Models;
+using NexusMods.Games.FOMOD;
+using NexusMods.Hashing.xxHash3;
+using NexusMods.MnemonicDB.Abstractions;
+using NexusMods.Paths;
+using NexusMods.Paths.Extensions;
+#pragma warning disable CS1591 // Missing XML comment for publicly visible type or member
+
+namespace NexusMods.Collections;
+using CollectionMod = Mod;
+
+[PublicAPI]
+public class InstallCollectionDownloadJob : IJobDefinitionWithStart<InstallCollectionDownloadJob, LoadoutItemGroup.ReadOnly>
```

Comment:
`InstallCollectionDownloadJob` is a bit unfortunate as a name but I understand the reasoning